runtime/jcl/common/reflecthelp.c

```diff
@@ -113,7 +113,11 @@ getClassAnnotationData(struct J9VMThread *vmThread, struct J9Class *declaringCla
 	j9object_t result = NULL;
 	U_32 *annotationData = getClassAnnotationsDataForROMClass(declaringClass->romClass);
 	if (NULL != annotationData) {
+		J9Class **classPtrAddr = 0;
+		U_32 byteCount = *annotationData;
 		result = getAnnotationDataAsByteArray(vmThread, annotationData);
+		classPtrAddr = (J9Class **)(I_8 *)J9JAVAARRAY_EA(vmThread, result, byteCount, I_8);
+		*classPtrAddr = declaringClass;
```

Comment:
likewise here, this needs to be written with the correct size. If compresedRefs `U_32` otherwise `UDATA`.